src/docfx/publish/PublishModelBuilder.cs

```diff
@@ -156,5 +218,57 @@ private bool IsInsideOutputFolder(string path)
             var outputFilePath = PathUtility.NormalizeFolder(Path.Combine(_outputPath, path));
             return outputFilePath.StartsWith(_outputPath);
         }
+
+        private enum ConflictingType
+        {
+            None,
+            OutputPathConflicts,
+            PublishUrlConflicts,
+        }
+
+        private class PublishItemKey : IEquatable<PublishItemKey>
+        {
+            public string[] Monikers { get; } = Array.Empty<string>();
+
+            public string? OutputPath { get; }
+
+            public string SiteUrl { get; } = "";
+
+            public FilePath FilePath { get; }
+
+            public PublishItemKey(PublishItem item, FilePath file)
+            {
+                Debug.Assert(item != null);
```

Comment:
With nullable enabled, can remove this line
```suggestion
                ~~~Debug.Assert(item != null);~~~
```